src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/RuntimeLoaderPass.php

```diff
@@ -33,7 +33,17 @@ public function process(ContainerBuilder $container): void
             $def = $container->getDefinition($id);
             $mapping[$def->getClass()] = new Reference($id);
         }
+        $attributesMapping = [];
+        foreach ($container->findTaggedServiceIds('twig.attribute_extension', true) as $id => $attributes) {
+            $def = $container->getDefinition($id);
+            $attributesMapping[$def->getClass()] = new Reference($id);
```

Comment:
shouldn't we skip this registration if `$mapping[$def->getClass()]` is set?

Reply:
I don't get your suggestion.

Classes with `RuntimeExtensionInterface` need to be injected into the runtime loader. Classes with `AsTwigExtension` need to be injected into the runtime loader, but also into the AttributeExtension. That's why there is 2 lists.

Reply:
but if a class is registered both using the regular way and using the attribute, we should skip registering it with the attribute

Reply:
Ah, correct. I don't expect anyone to use both the attribute and the interface on the same class, but this is not incompatible.

The + operator will remove duplicates.

Reply:
I would still skip here because of L46

Reply:
I've made a small modification to have the list of classes distinct from the class => service mapping. It ends to the same thing.